front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -397,6 +353,74 @@ export function DataSourceViewSelector({
   // Show the checkbox by default. Hide it only for tables where no child items are partially checked.
   const hideCheckbox = readonly || (isTableView && !isPartiallyChecked);
 
+  const getNodesFromConfig = (
+    selectionConfiguration: DataSourceViewSelectionConfiguration
+  ) =>
+    selectionConfiguration.selectedResources.reduce<
+      Record<string, ContentNodeTreeItemStatus>
+    >(
+      (acc, r) => ({
+        [r.internalId]: {
+          isSelected: true,
+          node: r,
+          parents: r.parentInternalIds || [],
+        },
+        ...acc,
+      }),
+      {}
+    );
+
+  const selectedNodes = getNodesFromConfig(selectionConfiguration);
+
+  const setSelectedNodes = useCallback(
+    (updater: TreeSelectionModelUpdater) => {
+      setSelectionConfigurations((prevState) => {
+        const prevSelectionConfiguration =
+          prevState[dataSourceView.sId] ??
+          defaultSelectionConfiguration(dataSourceView);
+        const selectedNodes = updater(
+          getNodesFromConfig(prevSelectionConfiguration)
+        );
+        const udpatedConfig = {
+          ...prevSelectionConfiguration,
+          selectedResources: Object.values(selectedNodes)
+            .filter((v) => v.isSelected)
+            .map((v) => ({
+              ...v.node,
+              parentInternalIds: v.parents,
+            })),
+          isSelectAll: false,
+        };
+
+        if (
+          udpatedConfig.selectedResources.length === 0 &&
+          !udpatedConfig.isSelectAll
```

Comment:
`!udpatedConfig.isSelectAll` will always be `true` as we explicitly set is to `false` above, no?